src/ScrapingFunctions.re

```diff
@@ -29,9 +37,58 @@ let validateNonNull = x =>
 
 let toStringProcess = x => Js.String.make(x);
 
+/** return the date x days ago in the format yyyy-MM-dd */
+let daysAgoDate = (x: string) => {
+  let now = Js.Date.make();
+
+  /** ISO date format is yyyy-MM-ddThh:mm:ss:msms.nnnZ, we only need the first 10 characters*/
+  let startIsoOfset = 0;
+  let lengthIsoOfset = 10;
+
+  switch (Js.Date.getDate(now) -. float_of_int(int_of_string(x))) {
+  | delta => Js.Date.setDate(now, delta)
+    |> Js.Date.fromFloat
+    |> Js.Date.toISOString
+    /** unfortunately Js.Date has no method to obtain format yyyy-MMM-dd
+       so we obtain the ISO format and extract the substring of interest */
+    |> Js.String.substrAtMost(~from=startIsoOfset, ~length=lengthIsoOfset);
+  | exception (Js.Exn.Error(err)) =>
+      Js.Exn.message(err)
+      |> Js.Option.getWithDefault("Unable to cast string to date")
+      /** trick to be able to log while still returning a string */
```

Comment:
why is it necessary to return a string?

Reply:
I think you could just use failwith in this case

Reply:
`daysAgoDate` takes a string representing a number (e.g. "5") and returns a string in date format.
If the casting of the string to an int fails it will return an empty string.
I figured that would make the function have no side effects.

If I `failwith` or just `Js.log` then the compiler will complain saying that it must return a string.

I'm open to modifying the return type of the fn if you think is more appropriate to do so